python-backend/app/api/parties/party/models/party.py

```diff
@@ -81,24 +96,71 @@ def search_by_name(cls, search_term, party_type=None, query_limit=50):
             return cls.query.filter(_filter_by_name).limit(query_limit)
 
     @classmethod
-    def create_party(cls, generated_first_name, generated_last_name, user_kwargs, save=True):
+    def create(cls,
+               # Required fields
+               party_name,
+               email,
+               phone_no,
+               party_type_code,
+               user_kwargs,
+               # Nullable fields
+               first_name=None,
+               phone_ext=None,
+               suite_no=None,
+               address_line_1=None,
+               address_line_2=None,
+               city=None,
+               province_code=None,
+               postal_code=None,
+               save=True):
         party = cls(
+            # Required fields
             party_guid=uuid.uuid4(),
-            first_name=generated_first_name,
-            party_name=generated_last_name,
-            email=generated_first_name.lower() + '.' + generated_last_name.lower() + '@' +
-            generated_last_name.lower() + '.com',
-            phone_no='123-123-1234',
-            party_type_code=PARTY_STATUS_CODE['per'],
-            **user_kwargs)
+            party_name=party_name,
+            email=email,
+            phone_no=phone_no,
+            party_type_code=party_type_code,
+            **user_kwargs,
+            # Optional fields
+            first_name=first_name,
+            phone_ext=phone_ext,
+            suite_no=suite_no,
+            address_line_1=address_line_1,
+            address_line_2=address_line_2,
+            city=city,
+            province_code=province_code,
+            postal_code=postal_code)
         if save:
             party.save(commit=False)
         return party
 
+
+    def validate_unique_name(self, party_name, first_name=None):
+        if Party.find_by_name(party_name, first_name):
+            if first_name:
+                name = first_name + ' ' + party_name
+            else:
+                name = party_name
+            raise AssertionError(
+                'Party with the name: {} already exists'.format(name))
+        return party_name
+
+    @validates('party_type_code')
+    def validate_email(self, key, party_type_code):
```

Comment:
Function name here seems off.

Reply:
Yep! That'll need to be fixed. These validate functions were moved around quite a bit while refactoring. Thanks for catching that